### ISSUE TITLE
Find a better way to handle 0/0 or x/0 in factor divide

### ISSUE THREAD

**Issue by ankurankan** [pgmpy/pgmpy#441]

**Comment by yashu-seth** [comment]
@ankurankan Please elaborate on this.

**Comment by ankurankan** [comment]
@yashu-seth Check these lines https://github.com/pgmpy/pgmpy/blob/dev/pgmpy/factors/Factor.py#L632. Here at line 634 we are specifically replacing the values with 0. We would ideally like to handle this within the divide in Line 630 only. So, we need to see if is this possible in some way, if yes then implement it.

**Comment by yashu-seth** [comment]
@ankurankan From what I understood, a 0/0 case will always give a nan . So I think we must have to replace it with 0 separately.  But we can put it in the same line 630 instead of doing it in 634. We can probably use this - 
<code>phi.values = np.array(list(0 if np.isnan(phi.values[index]/phi1.values[index]) else phi.values[index]/phi2.values[index] for index,value in enumerate(phi.values)))</code>

**Comment by ankurankan** [comment]
@yashu-seth No, using a for loop is too expensive. Will need to find a better way to do this.

**Comment by yashu-seth** [comment]
@ankurankan Okay, i'll see if I get a better solution.

**Comment by yashu-seth** [comment]
@ankurankan Can we use 

``` python
phi.values = np.nan_to_num(phi.values / phi1.values)
```

**Comment by ankurankan** [comment]
@yashu-seth I tried this but it's much slower than the current implementation.
